gitlet.js

```diff
@@ -813,6 +813,7 @@ var refs = {
     return ref !== undefined &&
       (ref.match("^refs/heads/[A-Za-z-]+$") ||
        ref.match("^refs/remotes/[A-Za-z-]+/[A-Za-z-]+$") ||
+       ref.match("^refs/remotes/[A-Za-z-]+\\\\[A-Za-z-]+$") ||
        ["HEAD", "FETCH_HEAD", "MERGE_HEAD"].indexOf(ref) !== -1);
   },
 
```

Comment:
I think this might not quite get to the root of the problem.  I assume that this line fixes the problem where a ref like `refs/remotes/origin\\master` is passed that has come from a Windows file path.  I'd rather normalise all the refs to use `/` as a path separator and convert refs that come from a Windows file path when they come into Gitlet.